src/Symfony/Component/Mailer/Transport/SendmailTransport.php

```diff
@@ -99,6 +100,18 @@ protected function doSend(SentMessage $message): void
 
         $this->stream->setCommand($command);
         $this->stream->initialize();
+        if (false !== strpos($command, ' -t')) {
+            $email = $message->getOriginalMessage();
+            if ($email instanceof Email) {
+                foreach ($email->getBcc() as $recipient) {
+                    $this->stream->write('Bcc:'.$recipient->toString()."\n");
```

Comment:
Is it allowed to send `Bcc: ` header BEFORE the rest of the payload?

Reply:
header now starts with `From:` and as `Bcc:` is part of the *destination fields* I don't think that should be a problem https://tools.ietf.org/html/rfc2822#section-3.6.3

But my main concern is that you "fix" this now on this place. But as `Bcc` is part of the official RFC for defining mails why is is not just part of the header of the message? And removed on the place where i'ts processed as described in the RFC?

So remove https://github.com/symfony/symfony/blob/d7c4433bee1079d2b93e48c1e36e5c92ff046fe0/src/Symfony/Component/Mime/Message.php#L101 and adjust the transports that handle the recipients self like smtp

Reply:
Sendmail and Smtp (and many other transports) send the exact same payload. Reverting this change will affect everybody, it would be BC break, and maybe a security issue.

The question is. Should we re-add the Bcc header for transport that support it? 
Or should we remove the Bcc header for transports that does not support it?

In my opinion, the option 1 (the one Symfony is currently using) is safe by default and we should keep that behavior.
Some transports (like Smtp, Sqs) provides alternatives ways to inject the recipients without sending this header.

Reply:
I get the BC issue. But currently the `Mime/Message.php` output is just not following the email specifications :grimacing: 

And yes the order transports handle this currently on there own. But IMO the code should try to be as close to the specifications to keep it understandable for others.

The other only "save" way to fix this for the sendMailTransport is to grep the raw message and search for the bcc header but not sure if that will be robust enough.

But adding every recipient as Bcc is not a real solution IMHO